cmd/kubermatic-installer/main.go

```diff
@@ -21,39 +21,70 @@ import (
 	"os"
 	"time"
 
-	"github.com/urfave/cli"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
 
 	"k8c.io/kubermatic/v2/pkg/log"
 	kubermaticversion "k8c.io/kubermatic/v2/pkg/version/kubermatic"
 )
 
-var (
-	verboseFlag = cli.BoolFlag{
-		Name:  "verbose, v",
-		Usage: "enable more verbose output",
-	}
+type Options struct {
+	Verbose         bool
+	ChartsDirectory string
+}
 
-	chartsDirectoryFlag = cli.StringFlag{
-		Name:   "charts-directory",
-		Value:  "charts",
-		Usage:  "filesystem path to the Kubermatic Helm charts",
-		EnvVar: "KUBERMATIC_CHARTS_DIRECTORY",
-	}
-)
+func (o *Options) CopyInto(other *Options) {
+	other.ChartsDirectory = o.ChartsDirectory
+	other.Verbose = o.Verbose
+}
+
+var options = &Options{}
 
 func main() {
 	rand.Seed(time.Now().UnixNano())
 
 	logger := log.NewLogrus()
 	versions := kubermaticversion.NewDefaultVersions()
 
-	app := cli.NewApp()
-	app.Name = "kubermatic-installer"
-	app.Usage = "Installs and updates Kubermatic Kubernetes Platform"
-	app.Version = versions.Kubermatic
-	app.HideVersion = true
-	app.Flags = flags()
-	app.Commands = commands(logger, versions)
+	rootCmd := &cobra.Command{
+		Use:           "kubermatic-installer",
+		Short:         "Installs and updates Kubermatic Kubernetes Platform",
+		Version:       versions.Kubermatic,
+		SilenceErrors: true,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			if env := os.Getenv("KUBERMATIC_CHARTS_DIRECTORY"); env != "" {
+				options.ChartsDirectory = env
+			}
```

Comment:
Isn't this going to override the `--charts-directory` flag? I think the flag should have precedence over the environment variable. Unless it worked the same way before, in that case, we should leave it as it is.

Reply:
I would not know how to write the code differently. We want to have the flag default to "charts", but the env must take precedence, as we use it in our Docker image to override the default location (https://github.com/kubermatic/kubermatic/blob/master/Dockerfile#L18).